tasks/prepare-fbc-release/prepare-fbc-release.yaml

```diff
@@ -0,0 +1,114 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: prepare-fbc-release
+  labels:
+    app.kubernetes.io/version: "1.0.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task for prepration of FBC release this task will
+    collect components from snapshot and set OCP version by
+    inspecting containerImage for each component.
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the Snapshot spec in the data workspace
+      type: string
+      default: "snapshot_spec.json"
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+      default: "data.json"
+  workspaces:
+    - name: data
+      description: Workspace where the snapshot is stored
+  steps:
+    - name: prepare-fbc-release
+      image: quay.io/hacbs-release/release-utils:a918b3a7cbd40add5955a2f1acb4c3dbb29c44e9
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+        if [ ! -f "${SNAPSHOT_PATH}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi        
+        
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        # Verify data.json exists
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi        
+
+        pattern="^v[0-9]+\.[0-9]+$"
+
+        # Read components and initial values
+        components=$(jq -c '.components[]' "$SNAPSHOT_PATH")
+        fromIndex=$(jq -r '.fbc.fromIndex' "$DATA_FILE")
+        targetIndex=$(jq -r '.fbc.targetIndex' "$DATA_FILE")
+        binaryImage=$(jq -r '.fbc.binaryImage' "$DATA_FILE")
+
+        # Print initial values
+        echo "Initial fromIndex: $fromIndex"
+        echo "Initial targetIndex: $targetIndex"
+        echo "Initial binaryImage: $binaryImage"
+        echo
+
+        # Count components
+        COMPONENT_COUNT=$(jq '.components | length' "$SNAPSHOT_PATH")
+        echo "found $COMPONENT_COUNT components"
+
+        # Initialize jq_script
+        jq_script="."
+
+        # Function to replace tag in an image
+        replace_tag() {
+            local updatedImage="${1%:*}:${2}"
+            echo "$updatedImage"
+        }
+
+        # Iterate over components
+        while read -r component; do
+            containerImage=$(echo "$component" | jq -r '.containerImage')
+            componentName=$(echo "$component" | jq -r '.name')
+
+            # Fetch OCP version using skopeo
+            ocpVersion=$(skopeo inspect --raw docker://"$containerImage" \
```

Comment:
Consider the scenario where skopeo inspect fails (e.g. Quay is down). This would likely just silently return an empty string. We should really fail in such case.

So I would suggest setting `set -o pipefail` at the top of the script.

Reply:
I agree. Thanks for the suggestion.

Reply:
OK, but why was `set -eu` removed at the same time? I think it's good to have as well.